src/app/pages/reveal/reveal.component.tsx

```diff
@@ -0,0 +1,134 @@
+import { Component, h } from 'preact';
+import LayoutGrid from 'preact-material-components/LayoutGrid';
+import './reveal.scss';
+import { Translate } from '../../components/translate/translate.component';
+import Button from 'preact-material-components/Button';
+import { GenericAccount } from 'moonlet-core/src/core/account';
+import Chips from 'preact-material-components/Chips';
+import { removeType } from '../../utils/remove-type';
+
+interface IProps {
+    account: GenericAccount;
+    words: string[];
+    topBarMiddleText: string;
+}
+
+interface IState {
+    wordsVisible: boolean;
+    layoutProps?: {
+        warningText: string;
+        tipsText: string;
+        revealButtonText: string;
+    };
+}
+
+export class RevealPage extends Component<IProps, IState> {
+    private wordsTextarea;
+    private isPrivateKeyLayout: boolean;
+
+    constructor(props: IProps) {
+        super(props);
+        let layoutProps;
+
+        switch (this.props.topBarMiddleText) {
+            case 'Reveal Secret Phrase':
```

Comment:
this check is not safe, we will translate texts from topbar, you can use route props check transaction-details page,
you cand check the current path.

Reply:
Done, thanks